### DISCUSSION TITLE
when ues  async_mode='eventlet' client can't get the message from server.

### DISCUSSION THREAD

**Discussion by 2651084156** [miguelgrinberg/Flask-SocketIO#1700] [Q&A]
i use code like this to send message 
<details>

  <summary>code</summary>

def ioioio(acc):
    a = 222
    v =111
    '''miaoshu'''
    time.sleep(1)
    '''def client_1():
        print('**************************')
        emit('server_response',
             {'data': pachong.idnumget['data'], 'data2': pachong.idnumget['data2'], 'data3': pachong.idnumget[acc]},
             broadcast=True)'''

    #print(pachong.idnumget[pachong.idnumget[acc]])
    #print(pachong.idnumget[pachong.idnumget[acc] + 'L'])
    #print('AAAAA')
    socketio.emit('server_response',{'data': str(listas), 'data2':a , 'data3': a},broadcast=True,namespace=name_space)
    socketio.emit('server_response',{'data': str(listas1), 'data2': v, 'data3': v}, broadcast=True, namespace=name_space1)

def getlong():
    time.sleep(4)
    while True:

       # print('+++++++++++++++')
        #print('AAAA')
        ioioio('AAA')

if __name__ == '__main__':
    t11 = threading.Thread(target=getlong, args=())
    print('111111++++++++++++++++++')
    t11.setDaemon(True)
    # socketio.start_background_task(target=getlong,kwargs=())
    t11.start()
    socketio.run(app, host='0.0.0.0', port=5000,debug=True )

</details>

but when i use async_mode='threading')  it work
but when use it it can not use websocket
and i want ues websocket
so what can i to slove it

**Comment by miguelgrinberg** [comment]
Are you reading the documentation?

If you stay with threading, you can install `simple-websocket` to add WebSocket support.
If you prefer eventlet, then you have to use eventlet's own concurrency functions, or else monkey patch the standard library.

**Reply by 2651084156** [reply]
> 
> 
> Are you reading the documentation?
> 
> If you stay with threading, you can install `simple-websocket` to add WebSocket support. If you prefer eventlet, then you have to use eventlet's own concurrency functions, or else monkey patch the standard library.

![图片](https://user-images.githubusercontent.com/69307837/137567098-6a7f4e8e-b6b3-4891-b9c9-548669bdb94d.png)
i try to ues this ,but when i try use is like threading like this
![图片](https://user-images.githubusercontent.com/69307837/137567208-a38a2288-ee74-463a-aed6-a4dcb62cb12b.png)
 it will give me a error like this 
![图片](https://user-images.githubusercontent.com/69307837/137567265-8bce7553-f696-4de1-8a9f-89c0bd0dd096.png)
i don know why because when use threading it can use

**Reply by miguelgrinberg** [reply]
Once again, you need to read the documentation to find the correct way to do things: https://flask-socketio.readthedocs.io/en/latest/api.html#flask_socketio.SocketIO.start_background_task

This function does not have an `args` argument, it uses standard positional and keyword arguments.

**Reply by 2651084156** [reply]
> 
> 
> Once again, you need to read the documentation to find the correct way to do things: https://flask-socketio.readthedocs.io/en/latest/api.html#flask_socketio.SocketIO.start_background_task
> 
> This function does not have an `args` argument, it uses standard positional and keyword arguments.

![图片](https://user-images.githubusercontent.com/69307837/137592631-bb279774-6960-4510-a554-7436f91bfda0.png)
will when i use kwargs is still give me a error.Is any thing wrong i do?

**Reply by miguelgrinberg** [reply]
You are doing it wrong, there are no `args` or `kwargs` arguments. See https://realpython.com/python-kwargs-and-args/ if you don't know how these arguments work.
